# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: xpcom/base/nsSystemInfo.cpp
# Commit: ac548c2a725b
# Full Hash: ac548c2a725b66db57c81bb24accca2bbe075931
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2019-07-29 21:38:39
# Description:
#   Bug 1565930, try to make Promise less error prone to compartment mismatches, r=bzbarsky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38762
# ==============================================================================

diff -r c4980c083852 -r ac548c2a725b xpcom/base/nsSystemInfo.cpp
--- a/xpcom/base/nsSystemInfo.cpp	Mon Jul 29 08:00:35 2019 +0000
+++ b/xpcom/base/nsSystemInfo.cpp	Mon Jul 29 14:43:54 2019 +0000
@@ -1171,7 +1171,7 @@
         }
 
         JS::Rooted<JS::Value> val(cx, JS::ObjectValue(*jsInfo));
-        capturedPromise->MaybeResolve(cx, val);
+        capturedPromise->MaybeResolve(val);
       },
       [capturedPromise](const nsresult rv) {
         capturedPromise->MaybeReject(rv);
